### PR TITLE
Fix empty container block check

### PR DESCRIPTION
**Pull request by islemaster**
- Removing hasEmptyContainerBlocks (was only used in one place, largely redundant, and somewhat dangerous that it was using a different algorithm than getEmptyContainerBlock)
- Extracting checkForEmptyContainerBlockFailure_, which happens to be the only place we look for empty container blocks.
- Putting the new method under test.

### CODE REVIEW

File: apps/src/feedback.js [hunk 1]
```diff
@@ -790,6 +781,28 @@ FeedbackUtils.prototype.getEmptyContainerBlock_ = function() {
...
+    if (type === 'procedures_defnoreturn' || type === 'procedures_defreturn') {
+      return TestResults.EMPTY_FUNCTION_BLOCK_FAIL;
+    }
+    // Block is assumed to be "if" or "repeat" if we reach here.
+    // This is where to add checks if you want a different TestResult
+    // for "controls_for_counter" blocks, for example.
+    return TestResults.EMPTY_BLOCK_FAIL;
+  }
+  return TestResults.ALL_PASS;
+};
```
Comment (Bjvanminnen) [inline apps/src/feedback.js]: Nit: could invert this: if (!emptyBlock) { return ALL_PASS; } and remove a level of indentation.

File: apps/test/feedbackTests.js [hunk 2]
```diff
@@ -13,6 +13,103 @@ global.window = {};
...
+    checkResultForBlocks({
+      result: TestResults.ALL_PASS,
+      blockXml: ''
+    });
+  });
+
+  it ("returns ALL_PASS when no container blocks are present", function () {
+    checkResultForBlocks({
+      result: TestResults.ALL_PASS,
+      blockXml: '<xml><block type="text_print"></block></xml>'
```
Comment (Bjvanminnen) [inline apps/test/feedbackTests.js]: No need to convert this, but FYI going forward there are a few utility functions in block_utils that make some of this xml generation easier (you can do stuff like blockUtils.blockOfType('textPrint'))

**Comment by Bjvanminnen** [comment]
lgtm. nice testing!

File: apps/test/feedbackTests.js (hunk 2 above)
Comment (islemaster) [inline apps/test/feedbackTests.js]: Ooh, good to know.  I was just going to levelbuilder to generate needed XML.  Oops.

Comment (Bjvanminnen) [inline apps/test/feedbackTests.js]: Another way to get the xml without going to levelbuilder. (you can also find this line in the comment at the top of levelTests.js).
```
Blockly.Xml.domToText(Blockly.Xml.blockSpaceToDom(Blockly.mainBlockSpace));

**Pull request merged**